.github/workflows/cloud-deploy-reusable-app-service-deploy.yml

```diff
@@ -0,0 +1,229 @@
+        required: true
+        type: string
+      docker_image:
+        required: true
+        type: string
+      sentry_project:
+        required: false
+        type: string
+      New_Relic_GUID:
+        required: false
+        type: string
+      deploy_sentry_release:
+        required: false
+        default: false
+        type: boolean
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+
+  infrastructure_data:
+    runs-on: ubuntu-latest
+    timeout-minutes: 80
+    environment: ${{ inputs.environment }}
+    env:
+      TF_WORKSPACE: ${{ inputs.terraform_workspace }}
+    permissions:
+      contents: read
+      deployments: write
+    outputs:
+      ecs_container_name: ${{ steps.terraform.outputs.ecs_container_name }}
+      ecs_service: ${{ steps.terraform.outputs.ecs_service }}
+      ecs_cluster: ${{ steps.terraform.outputs.ecs_cluster }}
+      task_name: ${{ steps.terraform.outputs.task_name }}
+      aws_region: ${{ steps.terraform.outputs.aws_region }}
+      services_to_deploy: ${{ steps.terraform.outputs.queue_workers_services }}
+    steps:
+      - run: echo "Deploying ${{ inputs.service_name }} to ${{ inputs.terraform_workspace }} And Docker Tag ${{ inputs.docker_image }}"
+
+      - name: Checkout cloud infra
+        uses: actions/checkout@v4
+        with:
+          repository: novuhq/cloud-infra
+          token: ${{ secrets.GH_PACKAGES }}
+          path: cloud-infra
+
+      - name: Terraform setup
+        uses: hashicorp/setup-terraform@v3
+        with:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+          terraform_version: 1.5.5
+          terraform_wrapper: false
+
+      - name: Terraform Init
+        working-directory: cloud-infra/terraform/novu/aws
+        run: terraform init
+
+      - name: Replace dash with underscore in service_name
+        id: replace_dash
+        run: echo "service_name_under=$(echo '${{ inputs.service_name }}' | tr '-' '_')" >> $GITHUB_ENV
+
+      - name: Terraform get output
+        working-directory: cloud-infra/terraform/novu/aws
+        id: terraform
+        env:
+          SERVICE_NAME: ${{ env.service_name_under }}
+        run: |
+          echo "ecs_container_name=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_container_name | jq -r .)" >> $GITHUB_OUTPUT
+          echo "ecs_service=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_service | jq -r .)" >> $GITHUB_OUTPUT
+          echo "ecs_cluster=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_cluster | jq -r .)" >> $GITHUB_OUTPUT
+          echo "task_name=$(terraform output -json ${{ env.SERVICE_NAME }}_task_name | jq -r .)" >> $GITHUB_OUTPUT
+          echo "aws_region=$(terraform output -json aws_region | jq -r .)" >> $GITHUB_OUTPUT
+          echo "queue_workers_services=$(terraform output -json queue_workers_services)" >> $GITHUB_OUTPUT
+
+  deploy_service_worker:
+    if: inputs.service_name == 'worker'
```

Comment:
What is the difference between worker and other services?

Reply:
@SokratisVidros The difference is the worker service is not one service like the rest of the services. It's deployed as five different worker sets one for each queue in the system.
This is also something you will want to keep in mind if you do add a new queue/process to Novu as it will require a new worker service to be set up.

Thus the worker is different, and in all honestly, all other services should deploy like the worker service to keep it cleanly organized, however, the terraform query will still be different.

Reply:
If not mistaken, this should be part of the ECS (or TS definition), not part of the CI. It's a smell that the CI need to change everytime we change a queue.